File: GXpert/GXpert.Web/Modules/ServerTypes/Content/ContentRatingColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ContentRatingRow } from "./ContentRatingRow";

export interface ContentRatingColumns {
    Id: Column<ContentRatingRow>;
    Name: Column<ContentRatingRow>;
    Description: Column<ContentRatingRow>;
}

export class ContentRatingColumns extends ColumnsBase<ContentRatingRow> {
    static readonly columnsKey = 'Content.ContentRating';
    static readonly Fields = fieldsProxy<ContentRatingColumns>();
}

File: GXpert/GXpert.Web/Modules/Content/ContentRating/ContentRatingColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Content.Columns;

[ColumnsScript("Content.ContentRating")]
[BasedOnRow(typeof(ContentRatingRow), CheckNames = true)]
public class ContentRatingColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
   
}